backend/services/snapshot_service.py

```
@@ -0,0 +1,216 @@
+        
+        results = await asyncio.gather(*snapshot_tasks)
+        
+        # Check for failures
+        failed_snapshots = [result for result in results if not result[0]]
+        if failed_snapshots:
+            raise Exception(f"Failed to create {len(failed_snapshots)} snapshots")
+        
+        return results
+    
+    async def create_remote_snapshots(self, source_host_info: HostInfo, compose_dir: str, 
+                                    volumes: List[VolumeMount], timestamp: Optional[str] = None) -> List[Tuple[str, str]]:
```

Flake8 (7.2.0)

[error] continuation line under-indented for visual indent

(E128)